tensorboard/plugins/graph/graph_util.py

```diff
@@ -0,0 +1,136 @@
+"""Utilities for graph plugin."""
+from tensorboard.compat.proto.graph_pb2 import GraphDef
+from tensorboard.compat.tensorflow_stub import dtypes
+
+
+def keras_model_to_graph_def(keras_layer):
+  """Returns GraphDef representation of the model.
+
+  Note that it only supports models that implemented get_config().
+
+  Args:
+    keras_layer: A dict from Keras model.to_json()
+
+  Returns:
+    A GraphDef representation of layers.
+  """
+
+  def get_layers(keras_layer):
+    yield ('', keras_layer)
+    if keras_layer.get('config').get('layers'):
+      name_scope = keras_layer.get('config').get('name')
+      for layer in keras_layer.get('config').get('layers'):
+        for (sub_name_scope, sublayer) in get_layers(layer):
+          sub_name_scope = '%s/%s' % (
+              name_scope, sub_name_scope) if sub_name_scope else name_scope
+          yield (sub_name_scope, sublayer)
+
+  def scoped_name(name_scope, node_name):
+    if name_scope:
+      return '%s/%s' % (name_scope, node_name)
+    return node_name
+
+  model_inputs_to_layer = {}
+  model_name_to_output_layer = {}
+  g = GraphDef()
+
+  # Sequential model layers do not have "inbound_nodes".
+  # Must assume all nodes are connected linearly.
+  prev_node_name = None
+  for (name_scope, layer) in get_layers(keras_layer):
+    layer_config = layer.get('config')
+    node_name = scoped_name(name_scope, layer_config.get('name'))
+
+    if layer_config.get('layers'):
+      # Do not add models to GraphDef. Store information for models.
+      input_layers = layer_config.get('input_layers')
+      inbound_nodes = layer.get('inbound_nodes')
+      if input_layers and inbound_nodes:
+        # Parent is Functional and current model is Functional.
+        for (input_layer, inbound_node) in zip(input_layers, inbound_nodes):
+          input_layer_name = scoped_name(node_name, input_layer)
+          inbound_node_name = scoped_name(name_scope, inbound_node[0])
+          model_inputs_to_layer[input_layer_name] = inbound_node_name
+      elif inbound_nodes:
+        # Parent is Functional but current model is Sequential.
+        # Sequential model can take only one input. Make sure inbound to the
+        # model is linked to the first layer in the Sequential model.
+        prev_node_name = scoped_name(name_scope, inbound_nodes[0][0])
+      elif input_layers and prev_node_name:
+        # Parent is Sequential but current model is Functional.
+        for input_layer in input_layers:
+          input_layer_name = scoped_name(node_name, input_layer)
+          model_inputs_to_layer[input_layer_name] = prev_node_name
+
+      # Unlike the name, there is only one item in the output_layers.
+      if layer_config.get('output_layers'):
+        maybe_layers = layer_config.get('output_layers')
+        # Normalizae output_layers.
+        # - single output: an array of values
+        # - multi outputs: an array of arrays of values
+        layers = maybe_layers if isinstance(
```

Comment:
```suggestion
        layers = (maybe_layers if isinstance(maybe_layers[0], (list,))
                      else [maybe_layers])
```

Conforms to Google Python style and is easier to read.

Reply:
Can you please link to the style rule? It's rather hard to read like this.

Reply:
https://engdoc.corp.google.com/eng/doc/devguide/py/style/index.md?cl=head#211-conditional-expressions

"Each portion must fit on one line: true-expression, if-expression, else-expression. Use a complete if statement when things get more complicated."

Recommended:

one_line = 'yes' if predicate(value) else 'no'
slightly_split = ('yes' if predicate(value)
                  else 'no, nein, nyet')